app/controllers/ConfirmationPageController.scala

```diff
@@ -51,121 +51,120 @@ trait ConfirmationPageController extends ERSReturnBaseController with Authentica
-                      Logger.error(s"File data not found: ${checkResult.status}")
-                      Future(getGlobalErrorPage)
-                    }
-                  }
-                } recover {
-                  case e: Throwable => {
-                    Logger.error(s"ERS connector call failed: ${e.getMessage}")
-                    getGlobalErrorPage
+                  } recover {
+                    case e: Throwable =>
+                      Logger.error(s"ERS connector call failed: ${e.getMessage}")
+                      getGlobalErrorPage
                   }
                 }
               }
+            } recover {
+              case e: Throwable =>
+                Logger.error(s"Get all data from cache failed with exception ${e.getMessage}, timestamp: ${System.currentTimeMillis()}.")
+                getGlobalErrorPage
             }
           } recover {
-            case e: Throwable => {
-              Logger.error(s"Get all data from cache failed with exception ${e.getMessage}, timestamp: ${System.currentTimeMillis()}.")
+            case e: Throwable =>
+              Logger.error(s"Get bundle failed with exception ${e.getMessage}, timestamp: ${System.currentTimeMillis()}.")
               getGlobalErrorPage
-            }
           }
         } recover {
-          case e: Throwable => {
-            Logger.error(s"Get bundle failed with exception ${e.getMessage}, timestamp: ${System.currentTimeMillis()}.")
+          case e: Throwable =>
+            Logger.error(s"Get ersMetaData with exception ${e.getMessage}, timestamp: ${System.currentTimeMillis()}.")
             getGlobalErrorPage
-          }
         }
-      } recover {
-        case e: Throwable => {
-          Logger.error(s"Get ersMetaData with exception ${e.getMessage}, timestamp: ${System.currentTimeMillis()}.")
-          getGlobalErrorPage
-        }
-      }
-    } else {
-      val url: String = ExternalUrls.portalDomain
-      cacheUtil.fetch[ErsMetaData](CacheUtil.ersMetaData, schemeRef).flatMap { all =>
-        Logger.info(s"Preventing resubmission of confirmation page, timestamp: ${System.currentTimeMillis()}.")
-        Future(Ok(views.html.confirmation(sessionDateTimeSubmitted, sessionBundelRef, all.schemeInfo.taxYear, url)(request, context, implicitly)))
-      } recover {
-        case e: Throwable => {
-          Logger.error(s"Get ersMetaData with exception ${e.getMessage}, timestamp: ${System.currentTimeMillis()}.")
-          getGlobalErrorPage
+      } else {
+        val url: String = ExternalUrls.portalDomain
+        cacheUtil.fetch[ErsMetaData](cacheUtil.ersMetaData, schemeRef).flatMap { all =>
+          Logger.info(s"Preventing resubmission of confirmation page, timestamp: ${System.currentTimeMillis()}.")
+
+          Future(Ok(views.html.confirmation(requestObject, sessionDateTimeSubmitted, sessionBundelRef, all.schemeInfo.taxYear, url)(request, context, implicitly)))
+
+        } recover {
```

Comment:
If this recover stack is too tall (which we discussed) the obvious solution would be to split it into private methods based on the recover requirements.

That at least would make it easier to read. Ticket or Do.